### ISSUE TITLE
 Whitespace in secret data passed by envs will be removed

### ISSUE THREAD

**Issue by Graffine** [appleboy/drone-ssh#111]
I try to pass GCR json key by envs and use it for docker pull on target machine and it failed. It caused by the whitespace in data string will be removed.

Here is my pipeline:
   
    publish-delta:    
        image: appleboy/drone-ssh
        host: <host>
        username: <user>
        secrets: [ gcr_json_key ]
        envs: [ gcr_json_key ]
        script:
          - docker login -u _json_key -p "$(echo $GCR_JSON_KEY)" <private_docker_registery>
          - docker pull <private_docker_image>

Suggests that keeping original input data string and it should be work.

**Comment by appleboy** [comment]
@Graffine Could you try the following solution?

encode your `gcr_json_key` using `base64` command.

```
$ cat ios_key.pem | base64
```

Update the script command

```
- docker login -u _json_key -p "$(echo $GCR_JSON_KEY | base64 --decode)" <private_docker_registery>
```

**Comment by Graffine** [comment]
@appleboy It is OK if I encode secret with base64. Thanks.

But it is still weird if the secret shared by different pipeline steps and we should handle it in each steps. Do you have any suggestions about it?

**Comment by joshdk** [comment]
I (coincidentally) fixed this in https://github.com/appleboy/drone-ssh/pull/113, if anyone wants to take a look.

**Comment by appleboy** [comment]
fixed in #113
